src/pool/tests/srv_scrubbing_tests.c

```diff
@@ -0,0 +1,705 @@
+
+	assert_success(rc);
+
+	return 0;
+}
+
+static void
+extent_deleted_by_aggregation(void **state)
+{
+	struct sts_context *ctx = *state;
+
+	sts_ctx_update(ctx, 1, TEST_IOD_ARRAY_1, "dkey", "akey",
+		       1, true);
+
+	ctx->tsc_yield_fn = test_yield_deletes_extent;
+	ctx->tsc_sched_arg = ctx;
+
+	sts_ctx_do_scrub(ctx);
+
+	/* First epoch should not exist */
+	assert_rc_equal(sts_ctx_fetch(ctx, 1, TEST_IOD_ARRAY_1, "dkey",
+				     "akey", 1), -DER_NONEXIST);
+	/* Second (inserted by test_yield_deletes_extent) should now exist */
+	assert_success(sts_ctx_fetch(ctx, 1, TEST_IOD_ARRAY_1, "dkey",
+				     "akey", 2));
+
+}
+
+static void
+multiple_objects(void **state)
+{
+	struct sts_context *ctx = *state;
+
+	sts_ctx_update(ctx, 1, TEST_IOD_SINGLE, "dkey", "akey", 1, false);
+	sts_ctx_update(ctx, 2, TEST_IOD_SINGLE, "dkey", "akey", 1, false);
+	sts_ctx_update(ctx, 3, TEST_IOD_SINGLE, "dkey", "akey", 1, false);
+	sts_ctx_update(ctx, 4, TEST_IOD_SINGLE, "dkey", "akey", 1, true);
+
+	sts_ctx_do_scrub(ctx);
+
+	assert_success(sts_ctx_fetch(ctx, 1, TEST_IOD_SINGLE, "dkey",
+				     "akey", 1));
+	assert_success(sts_ctx_fetch(ctx, 2, TEST_IOD_SINGLE, "dkey",
+				     "akey", 1));
+	assert_success(sts_ctx_fetch(ctx, 3, TEST_IOD_SINGLE, "dkey",
+				     "akey", 1));
+	assert_csum_error(sts_ctx_fetch(ctx, 4, TEST_IOD_SINGLE, "dkey",
+					"akey", 1));
+}
+
+static int
+sts_setup(void **state)
+{
+	struct sts_context	*ctx;
+
+	D_ALLOC_PTR(ctx);
+
+	assert_non_null(ctx);
+	sts_ctx_init(ctx);
+	*state = ctx;
+
+	/* set some defaults */
+	ctx->tsc_pool.sp_scrub_sched = DAOS_SCRUB_SCHED_RUN_WAIT;
+	ctx->tsc_pool.sp_scrub_freq_sec = 1;
+	ctx->tsc_pool.sp_scrub_cred = 1;
+
+	return 0;
+}
+
+static int
+sts_teardown(void **state)
+{
+	struct sts_context *ctx = *state;
+
+	sts_ctx_fini(ctx);
+	D_FREE(ctx);
+
+	return 0;
+}
+
+#define	TS(desc, test_fn) \
+	{ desc, test_fn, sts_setup, sts_teardown }
+
+static const struct CMUnitTest scrubbing_tests[] = {
+	TS("CSUM_SCRUBBING_01: SV with no corruption",
+	   scrubbing_with_no_corruption_sv),
+	TS("CSUM_SCRUBBING_02: Array with no corruption",
+	   scrubbing_with_no_corruption_array),
+	TS("CSUM_SCRUBBING_03: A single value corrupted",
+	   scrubbing_with_sv_corrupted),
+	TS("CSUM_SCRUBBING_04: A corrupted extent value",
+	   corrupted_extent),
+	TS("CSUM_SCRUBBING_05: Multiple corrupted extent values",
+	   scrubbing_with_arrays_corrupted),
+	TS("CSUM_SCRUBBING_06: Scrub multiple epochs",
+	   scrub_multiple_epochs),
+	TS("CSUM_SCRUBBING_06: Multiple keys, some corrupted values",
+	   scrubbing_with_multiple_akeys),
+	TS("CSUM_SCRUBBING_07: Multiple writes to same key, then corruption",
+	   scrubbing_with_good_akey_then_bad_akey),
+	TS("CSUM_SCRUBBING_08: Extent is deleted during scrub while yielding",
+	   extent_deleted_by_aggregation),
+	TS("CSUM_SCRUBBING_09: Scrubbing multiple objects",
+	   multiple_objects),
+};
+
+extern int run_scrubbing_sched_tests();
```

Comment:
(style)  Bad function definition - int run_scrubbing_sched_tests() should probably be int run_scrubbing_sched_tests(void)
(style)  externs should be avoided in .c files